pg_shard.c

```diff
@@ -226,8 +226,18 @@ PgShardPlanner(Query *query, int cursorOptions, ParamListInfo boundParams)
 
 		ErrorIfQueryNotSupported(distributedQuery);
 
-		/* compute the list of shards this query needs to access */
+		/*
+		 * Compute the list of shards this query needs to access.
+		 * Error out if there are no existing shards for the table.
+		 */
 		queryShardList = DistributedQueryShardList(distributedQuery);
+		if (queryShardList == NIL)
+		{
+			ereport(ERROR, (errmsg("cannot plan SELECT query"),
```

Comment:
Also, the style guide says this [should be in past tense](http://www.postgresql.org/docs/9.4/static/error-style-guide.html#AEN104799): _could not plan query_.